doc/RELEASE.adoc

```diff
@@ -0,0 +1,132 @@
+= Plutus Release Process
+:toc: left
+:reproducible:
+:figure-caption!:
+
+== Packages
+
+The following packages are versioned:
+
+- Libraries:
+* plutus-core:plutus-core
+* plutus-ledger-api:plutus-ledger-api
+* plutus-tx:plutus-tx
+* plutus-tx-plugin:plutus-tx-plugin
+- Executables:
+* plutus-core:pir
+* plutus-core:plc
+* plutus-core:uplc
+
+== Version Scheme
+
+All above packages are versioned and released in sync.
+That is, whenever one package releases version X, so do all other packages.
+This greatly simplifies the release process compared to versioning and releasing the packages independently.
+
+We follow https://pvp.haskell.org/[the PVP version scheme] for the packages, where version numbers have a `major.major.minor.patch` pattern with the following semantics:
+
+- A major release (e.g., 3.5.2.1 -> 3.6.0.0 or 4.0.0.0) may contain arbitrary changes to the API (subject to the backwards compatibility requirements described below).
+- A minor release (e.g., 3.5.2.1 -> 3.5.3.0) may contain changes such as new features, which technically can be breaking, but are unlikely to cause breakage in practice, and if it does, the fix should be straightforward (e.g., by renaming).
++
+A minor release is also allowed to change the observable behaviors of functions that users should not rely on.
+For example, if a function returns a list of things and makes no promise on the order of elements in the returned list, then a minor release may change the order.
+- A patch release (e.g., 3.5.2.1 -> 3.5.2.2) can only contain such things as simple bug fixes, performance improvements, and documentation updates.
+
+=== Version Qualifiers
+
+Version qualifiers, also known as pre-release versions, usually refer to alpha, beta, and release candidate.
+Typically, alpha and beta are pre-feature complete, while release candidates are post-feature complete.
+
+We do not tag or publish alpha or beta versions for libraries.
+Depending on an alpha or beta version of a library requires `source-repository-package`, which is quite inconvenient.
+It is also uncommon for a library to publish pre-release versions (unless it is new or a complete rewrite).
+However, we may annotate individual functions in a library as alpha or beta if we deem it appropriate.
+
+We do tag release candidates, e.g., 3.5.2.1-rc1, because additional non-CI tests and checks may need to be carried out.
+These tests and checks either are too expensive to run in CI, or require manual steps (e.g., security audit), but are prerequisites of making a release.
```

Comment:
Another place where we could just reference the Release QA process. Generally I'm pretty in favour of having dedicated sections for stuff like this and just cross-referencing a lot.